addons/point_of_sale/models/pos_session.py

```diff
@@ -1770,7 +1774,7 @@ def _get_pos_ui_res_partner(self, params):
         partner_ids = [res[0] for res in self.config_id.get_limited_partners_loading()]
         # Need to search_read because get_limited_partners_loading
         # might return a partner id that is not accessible.
-        params['search_params']['domain'] = [('id', 'in', partner_ids)]
+        params['search_params']['domain'] = AND([self._get_partners_domain(),[('id', 'in', partner_ids)]])
```

Comment:
remove this change as well. If you want a custom way of loading limited partners, you can just override `get_limited_partners_loading`.

Reply:
Dear @caburj,
Thank you for your comment. While we can override `get_limited_partners_loading`, we still need the `_get_partners_domain` field to save the custom domain for limiting partners. This field is necessary for our solution.

Reply:
hi @k700mx ,
@caburj is making things simple for you, in the stable versions, it is not allowed to introduce a new field in existing module, but in some cases, like as you have currently, there will people who needs to add the domain and filter the partners, in such cases they can easily inherit the newly adding hook in their custom module and achieve it, rather than rewriting the entire function.

i hope its clear for you.
thanks

Reply:
seems almost done @k700mx ,
try squashing all the commits and keep one for the cla and others for this feature, also commit message has to be in format [IMP] point_of_sale: Add partner domain filter